src/main/java/com/meilisearch/sdk/SearchRequest.java

```diff
@@ -94,7 +106,78 @@ public SearchRequest(
             String[] attributesToCrop,
             int cropLength,
             String[] attributesToHighlight,
-            String filter,
+            String[] filter,
+            boolean matches,
+            String[] facetsDistribution,
+            String[] sort) {
+        this(
+                q,
+                offset,
+                limit,
+                attributesToRetrieve,
+                attributesToCrop,
+                cropLength,
+                attributesToHighlight,
+                filter,
+                null,
+                matches,
+                facetsDistribution,
+                sort);
+    }
+
+    /**
+     * Full SearchRequest Constructor for building search queries with 2D filter Array
+     *
+     * @param q Query string
+     * @param offset Number of documents to skip
+     * @param limit Maximum number of documents returned
+     * @param attributesToRetrieve Attributes to display in the returned documents
+     * @param attributesToCrop Attributes whose values have been cropped
+     * @param cropLength Length used to crop field values
+     * @param attributesToHighlight Attributes whose values will contain highlighted matching terms
+     * @param filterArray String array that can take multiple nested filters
+     * @param matches Defines whether an object that contains information about the matches should
+     *     be returned or not
+     * @param facetsDistribution Facets for which to retrieve the matching count
+     * @param sort Sort queries by an attribute value
+     */
+    public SearchRequest(
+            String q,
+            int offset,
+            int limit,
+            String[] attributesToRetrieve,
+            String[] attributesToCrop,
+            int cropLength,
+            String[] attributesToHighlight,
+            String[][] filterArray,
+            boolean matches,
+            String[] facetsDistribution,
+            String[] sort) {
+        this(
+                q,
+                offset,
+                limit,
+                attributesToRetrieve,
+                attributesToCrop,
+                cropLength,
+                attributesToHighlight,
+                null,
+                filterArray,
+                matches,
+                facetsDistribution,
+                sort);
+    }
+
+    private SearchRequest(
```

Comment:
I added a `private` constructor that will be called from the two constructors that take `filter` and `filterArray`